### ISSUE TITLE
App crashes/closes when second server on same port opens

### ISSUE THREAD

**Issue by sayezz** [cutelyst/cutelyst#207]
I try to catch the case when a user tries to open a server on the same port where already a server is running.
This is the code I use so far:
`
int main(int argc, char *argv[])
{
    QCoreApplication app(argc, argv);

    auto myapp = new Server();
    auto wsgi = new CWSGI::WSGI;
    wsgi->setHttpSocket({
                            { QStringLiteral(":1234") },
                        });
    wsgi->setUpgradeH2c(true);
    wsgi->setBufferSize(16393);
    int ret = wsgi->exec(myapp); //the application closes here without having the time to set "ret"
}
`

When I run this snipped a second time I get an output on the console:

> Failed to listen on TCP: :1234 : The bound address is already in use

So far so good. The problem is that the application closes before I can handle this failuer. This leads to the fact that in my real life application the applications closes/crashes when a user tries to open two server on the same port.

Any suggestion how I can check wheter the port is still available?
Thanks

**Comment by sayezz** [comment]
I don't know if this is the right solution but here is whar works for me:

```
    QTcpServer *server;
    server = new QTcpServer();

    bool ret = server->listen(QHostAddress::Any, 1234);
    if (ret) {
        server->close();
        wsgi->setHttpSocket({
            { QStringLiteral(":1234") },
        });
        wsgi->setUpgradeH2c(true);
        wsgi->setBufferSize(16393);

        wsgi->exec(myapp);
    }
```

If there is a better solution, please let me know. Otherwise this can be closed.

**Comment by dantti** [comment]
This isn't the right solution because another server can start listening before Cutelyst starts listening, this needs to be fixed in WSGI module.

**Comment by sayezz** [comment]
Do I need to fix this on my side? Or is this something that must be fixed in Cutelyst?

**Comment by dantti** [comment]
No, it's just that I'm quite busy ATM, as soon as I fix this I'll close the issue

**Comment by sayezz** [comment]
You are doing a great job! Keep it up!

**Comment by dantti** [comment]
`Failed to listen on TCP: :1234 : The bound address is already in use`

This issue was fixed in 490d3fe1f3c4d24b384b38ca79605de51ac64a32

**Comment by dantti** [comment]
I think we now properly handle not closing the application when Cutelystd::start() faild can you confirm?
